__tests__/app.test.js

```diff
@@ -39,45 +39,64 @@ describe("GET /api/topics", () => {
-    });
-    
+describe("GET /api", () => {
+  it("200: responds with an object containing each available endpoint and description", () => {
+    return request(app)
+      .get("/api")
+      .expect(200)
+      .then(({ body: endpoints }) => {
+        const expectedResults = endpointsJsonFile;
+        expect(endpoints).toEqual(expectedResults);
+      });
+  });
 });
 
-describe('GET /api/articles/:article_id', () => {
-    it('200: responds with an object matching the requested id, with correct properties', () => {
-        return request(app)
-        .get('/api/articles/1')
-        .expect(200)
-        .then(({body:article}) => {
-            const expectedResult = endpointsJsonFile["GET /api/articles/:article_id"].exampleResponse.article
-            expect(article).toMatchObject(expectedResult)
-        })
-    });
-    it("404: responds with 'Not Found' when given valid but non-existing id", () => {
-      return request(app)
-      .get('/api/articles/987654321')
+describe("GET /api/articles/:article_id", () => {
+  it("200: responds with an object matching the requested id, with correct properties", () => {
+    return request(app)
+      .get("/api/articles/1")
+      .expect(200)
+      .then(({ body}) => {
+        const article = body.article
+        const expectedResult =
+          endpointsJsonFile["GET /api/articles/:article_id"].exampleResponse
+            .article;
+        expect(article).toMatchObject(expectedResult);
+      });
+  });
+  it("404: responds with 'Not Found' when given valid but non-existing id", () => {
+    return request(app)
+      .get("/api/articles/987654321")
       .expect(404)
-      .then(({body}) =>{
-        const errorMsg = body.msg
-        expect(errorMsg).toBe("Not Found")
-      })
-    })
-    it("400: responds with 'Bad Request' when failing schema validation", () => {
-      return request(app)
-      .get('/api/articles/notAValidId')
+      .then(({ body }) => {
+        const errorMsg = body.msg;
+        expect(errorMsg).toBe("Not Found");
+      });
+  });
+  it("400: responds with 'Bad Request' when failing schema validation", () => {
+    return request(app)
+      .get("/api/articles/notAValidId")
       .expect(400)
-      .then(({body}) =>{
-        const errorMsg = body.msg
-        expect(errorMsg).toBe("Bad Request")
+      .then(({ body }) => {
+        const errorMsg = body.msg;
+        expect(errorMsg).toBe("Bad Request");
+      });
+  });
+});
+
+describe('GET /api/articles', () => {
+  it('200: responds with array of article objects sorted descending by date created, with correct properties', () => {
+    return request(app)
+    .get("/api/articles")
+    .expect(200)
+    .then(({body}) =>{
+      const articles = body.articles
+      expect(articles).toHaveLength(13)
+      expect(articles).toBeSortedBy('created_at', {descending:true})
+      articles.forEach(article => {
+        expect(article).not.toHaveProperty('body')
       })
     })
-});
+  });
+  
```

Comment:
We should have a test to check all the properties of article including comment_count 🙂